src/main/kotlin/skytils/skytilsmod/utils/apiTools.kt

```diff
@@ -141,4 +141,22 @@ object BlockPosObjectSerializer : KSerializer<BlockPos> {
             }
             BlockPos(x, y, z)
         }
+}
+
+/**
+ * Serializes/deserializes a [SkyblockIsland] based on its [SkyblockIsland.mode] to retain compatibility
+ * with older saves that used the mode instead of a stringified enum constant name.
+ */
+object SkyblockIslandModeSerializer : KSerializer<SkyblockIsland> {
```

Comment:
i swear there's already a serializer for this

Reply:
Enum constants serialize perfectly fine without any extra effort, but I wanted to use the "mode" property of the SkyblockIsland for compatibility with old saves which use values like "mining_3" instead of "DwarvenMines"

Reply:
https://github.com/FluxCapacitor2/SkytilsMod/blob/9433871ec5109e0ee277ad5dfe8b9ccf98946192/src/main/kotlin/skytils/skytilsmod/utils/SBInfo.kt#L187

You can remove either one of them